plugins/topology/src/utils/resource-utils.ts

```diff
@@ -288,3 +291,55 @@ export const getUrlForResource = (
     getIngressURLForResource(resources, resource)
   );
 };
+
+const byCreationTime = (left: any, right: any): number => {
+  const leftCreationTime = new Date(
+    left?.metadata?.creationTimestamp || Date.now(),
+  );
+  const rightCreationTime = new Date(
+    right?.metadata?.creationTimestamp || Date.now(),
+  );
+  return rightCreationTime.getTime() - leftCreationTime.getTime();
+};
+
+const getPipelineRunsForPipeline = (
+  pipeline: any,
+  resources: K8sResponseData,
+): any[] => {
+  if (!resources.pipelineruns?.data?.length) {
+    return [];
+  }
+  const pipelineRunsData = resources.pipelineruns.data;
+  const PIPELINE_RUN_LABEL = 'tekton.dev/pipeline';
+  const pipelineName = pipeline.metadata.name;
+  return pipelineRunsData
+    .filter((pr: any) => {
+      return (
+        pipelineName ===
+        (pr.spec?.pipelineRef?.name ||
+          pr?.metadata?.labels?.[PIPELINE_RUN_LABEL])
+      );
+    })
+    .sort(byCreationTime);
+};
+
+export const getPipelinesAndPipelineRunsForResource = (
+  resource: K8sWorkloadResource,
+  resources: K8sResponseData,
+): { pipelines: any[]; pipelineRuns: any[] } | null => {
+  if (!resources.pipelines?.data?.length) {
+    return null;
+  }
+  const pipelinesData = resources.pipelines.data;
+  const resourceInstanceName =
+    resource.metadata?.labels?.[INSTANCE_LABEL] || null;
+  if (!resourceInstanceName) return null;
+  const resourcePipeline = pipelinesData.find(
+    pl => pl.metadata?.labels?.[INSTANCE_LABEL] === resourceInstanceName,
+  );
+  if (!resourcePipeline) return null;
+  return {
+    pipelines: [resourcePipeline],
```

Comment:
Why not

```suggestion
    pipeline: resourcePipeline,
```

Reply:
So currently we are just showing a single pipeline connected to the workload but their can be more than one pipeline connected to it. This is similar to ODC and at that time their might have been a plan to show multiple pipelines later.